safekeeper/src/timeline.rs

```diff
@@ -318,17 +286,13 @@ impl Timeline {
     /// Register compute connection, starting timeline-related activity if it is
     /// not running yet.
     /// Can fail only if channel to a static thread got closed, which is not normal at all.
-    pub fn on_compute_connect(&self, pageserver_connstr: Option<&String>) -> Result<()> {
+    pub fn on_compute_connect(&self) -> Result<()> {
         let is_wal_backup_action_pending: bool;
         {
             let mut shared_state = self.mutex.lock().unwrap();
             shared_state.num_computes += 1;
             is_wal_backup_action_pending = shared_state.update_status();
-            // FIXME: currently we always adopt latest pageserver connstr, but we
-            // should have kind of generations assigned by compute to distinguish
-            // the latest one or even pass it through consensus to reliably deliver
-            // to all safekeepers.
-            shared_state.activate_walsender(&self.zttid, pageserver_connstr.cloned());
+            shared_state.active = true;
```

Comment:
It seems that you don't need to change `shared_state.active` here and in `stop_walsender`, because `(de)activate_walsender` wasn't doing that. Maybe I'm missing something, @arssher take a look

Reply:
This is exactly why I need the review and why I'm not sure about the `active = true` purpose.

The WAL streaming decision could be affected two different ways now:
* from the safeeeker part, that filters inactive ones in broker.rs — only active timelines are being pushed into etcd
* from the safeekeeper part, when it adds/removes the pageserver_connstr for any timeline: only the timelines with pageserver_connstr are considered worthy of connection on pageserver side

Judging from the current code, some logic happens to `deactivate_walsender`, so there at least was a need to shut down walstream in a custom way and that's what I wanted to reimplement somehow: we have a single global broker push now, so there's no way to "disable" it per timeline without extra properties or `active = false`

Reply:
I also see that `active` was not really set anywhere explicitly?  A bit odd.